yw_website/yw_website/apps/website/utils.py

```diff
@@ -53,38 +53,48 @@ def get_block_data(run_id):
     '''
     program_blocks = ProgramBlock.objects.filter(run=run_id)
     data = Data.objects.filter(run=run_id)
+    # print(data)
 
     blocks = []
     data_vals = []
 
     # grab parent block
-    parent = ProgramBlock()
-    parent.name = program_blocks[0].name
-    parent.programblock_id = program_blocks[0].programblock_id
-    parent.id = program_blocks[0].id
-    parent.direct_descendents = program_blocks.filter(in_program_block=parent.id)
-    blocks.append(parent)
-
-    # prints for testing
-    # print(parent.name)
-    # print(parent.in_program_block_id)
-    # print(parent.direct_descendents)
-    print()
+    parents = get_direct_descendants(None, program_blocks)
+    print(parents)
+    # for data in parent_data:
+    #     parent.data_objs.append(data)
 
     # grab child blocks
-    for block in program_blocks[1:]:
+    child_blocks = program_blocks.exclude(in_program_block_id=None)
+    for block in child_blocks:
```

Comment:
the `get_direct_descendants` function is recursive and sets all the children already, you can remove this for loop